### DISCUSSION TITLE
INTERNAL_FUNCTION_INVOCATION_FAILED error on production

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#37764] [Help]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

    Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro
    Binaries:
      Node: 18.3.0
      npm: N/A
      Yarn: N/A
      pnpm: 7.1.9
    Relevant packages:
      next: 12.1.6
      react: 18.1.0
      react-dom: 18.1.0

### How are you deploying your application? (if relevant)

Vercel

### Describe the Bug

I got this error message in an API endpoint: 
Vercel's function's log: 
`INTERNAL_FUNCTION_INVOCATION_FAILED: A server error has occurred`

API function returned: 
`{"error": {"code": "500", "message": "A server error has occurred"}}`

### Expected Behavior

When I work with the same API function in the development, it works just fine with all the environment variable.

### To Reproduce

We have a big repo, not sure how to reproduce this yet. I ask first because maybe this is some kind of internal function from vercel. Otherwise, I will try to reproduce later. Thanks

**Comment by ghost** [comment]
I would love to know when we have this problem, does it actually hits the application's API yet?

**Comment by ijjk** [comment]
Hi, can you share a deployment or a repo with a minimal reproduction where you are seeing this?

**Reply by ghost** [reply]
ok, I will ping you when I have it, doing it now.

**Reply by ghost** [reply]
@ijjk there is a problem in one of my imported modules from other packages which causes the problem. However, the thing is it works fine on development. 
Let me create the simplest reproducible version.

**Reply by ghost** [reply]
update on the progress, for some reason when I imported the module `const language = require("@google-cloud/language");`  in my current codebase, it causes the error.

However, when I created a brand new project, it works just fine. Might it be package conflict or something? I use `pnpm`.

Thanks

**Reply by ijjk** [reply]
Are there any errors showing in your function logs on the deployment?

**Reply by ghost** [reply]
@ijjk yes I got this error: 
```
[GET] /endpoint
00:35:19:72
INTERNAL_FUNCTION_INVOCATION_FAILED: A server error has occurred
```

whenever I import the package above, all the API endpoints get this kind of error, and I have tried to just return the `res.status(200).json({})`, the API handler is not hit. Something before it causes this error, is there anyway we get the trace? some how

**Reply by ijjk** [reply]
It sounds like the node process is crashing when this dependency is imported during initialization so it doesn't have a chance to fully setup. Does this dependency rely on native modules that may be incompatible with the AWS lambda environment?

**Reply by ghost** [reply]
yeah, it might, I will debug all the deps tomorrow to see which "combination" that causes the crash. Since if I import `"@google-cloud/language"` alone, everything works just fine.

**Reply by ghost** [reply]
@ijjk I finally found the problem, it might be something wrong with pnpm and vercel's initialization. 
You can check the repo with 2 simple branches ( already simplified ) here: https://github.com/rhacker/debug_vercel

The problem is when I import `@google-cloud/language` from a shared package in pnpm workspace, it just causes the error when requesting the API endpoint: `INTERNAL_FUNCTION_INVOCATION_FAILED: A server error has occurred
` . For the API package to work, I need to specify `@google-cloud/language` in the dep of the API package as well.

**Reply by ghost** [reply]
@ijjk it turns out, I need to install using `pnpm install -r` to install all the deps recursively, pnpm doesn't install deps of related workspace as default.

**Reply by wovalle** [reply]
Hi @rhacker, did you (or @ijjk) figure out why specially `@google-cloud/language` had this problem? I'm trying to understand why is happening to me as well (not with the same dependency). Same setup, pnpm monorepo but `pnpm install -r` didn't worked.

**Reply by martin-landhi** [reply]
Any other information about this issue INTERNAL_FUNCTION_INVOCATION_FAILED?

where can i debug it? because its happening to me and appear randomly, was not after some deploy etc

**Reply by ahazeemi** [reply]
@martin-landhi Seems an issue with vercel right now: https://www.vercel-status.com/. It is appearing for me as well

**Reply by Wyn-Price** [reply]
@ahazeemi thank you, I'm having the same issue too. Not sure if it's relevant but I'm using express

**Reply by franknoel** [reply]
The issue is caused by AWS:
https://health.aws.amazon.com/health/status

@ijjk what would be the best way to fallback this type of error on a site with a middleware and SSR ?

**Reply by ahazeemi** [reply]
For an immediate fix, please update your region to cle1 on Vercel and redeploy (https://vercel.com/docs/concepts/functions/serverless-functions/regions). Worked for me

**Reply by ojvribeiro** [reply]
As @ahazeemi said, it is a Vercel issue. I solved by changind the Function Region in settings to `cle1` and redeploying.

**Reply by sephorus** [reply]
This worked for me, thanks

**Reply by frankolson** [reply]
Updating the Function Region in settings to `cle1` and redeploying worked for me as well, thank you

**Reply by Abdulazeez-ojulari** [reply]
This worked for me thanks man

**Reply by reginbald** [reply]
Is there a way to get notified if this happens again so that I can change regions?

**Reply by ayazhussain1** [reply]
Following.

**Reply by franknoel** [reply]
Following the service health page on AWS and changing the functions region on Vercel is a temporary fix. Having a proper fallback page would be way better than showing an `INTERNAL_FUNCTION_INVOCATION_FAILED` error.

**Reply by frankolson** [reply]
Heads for any one using Cloudflare R2, this seems to cause issues when using the S3 sdk. When a region is not specified, `auto` will be the default and for some reason switching the function region to `cle1` was causing the S3 sdk region to `us-east-2` resulting in the following error when trying to perform `PUT` actions:
```
InvalidRegionName: The region name 'us-east-2' is not valid. Must be one of: wnam, enam, weur, eeur, apac, auto
```

Now that this issue has been resolved by Vercel, setting the function region back to Washington, D.C., USA (iad1) fixed the issue for me.
